### PR TITLE
Handle status code 400

### PR DESCRIPTION
**Pull request by gandarez**
This PR handles http status code 400 (Bad Request) and return default exit (1).

### CODE REVIEW

**Review by alanhamlett — COMMENTED** [review]
We need integration tests, because `errors.As(err, &errapi)` etc aren't matching error types correctly. Maybe because the error type is handled and re-returned as a different type before it gets to the `errors.As` line.

Either way, we need tests that check the exit code.

**Comment by alanhamlett** [comment]
I think this is the line that breaks matching error type, by re-returning the error as the wrong type:

https://github.com/wakatime/wakatime-cli/blob/0760642ff424b8a9402b5dfa9e498825ffc0fd69/cmd/legacy/heartbeat/heartbeat.go#L187

**Comment by gandarez** [comment]
@alanhamlett added an integration test to validate error when "bad request".

**Comment by alanhamlett** [comment]
> added an integration test to validate error when "bad request".

In the tests, we should make sure the exit code matches `exitcode.ErrAPI` or for `400` that it matches `exitcode.ErrDefault`.

**Comment by gandarez** [comment]
> > added an integration test to validate error when "bad request".
> 
> In the tests, we should make sure the exit code matches `exitcode.ErrAPI` or for `400` that it matches `exitcode.ErrDefault`.

Integration test now checks for exit code.

File: cmd/legacy/heartbeat/heartbeat.go [hunk 1]
```diff
@@ -42,11 +42,19 @@ func Run(v *viper.Viper) (int, error) {
 			)
 		}
 
+		var errbadRequest api.ErrBadRequest
+		if errors.As(err, &errbadRequest) {
+			return exitcode.ErrDefault, fmt.Errorf(
+				"failed to send heartbeat(s) due to api error: %s",
```
Comment (alanhamlett) [inline cmd/legacy/heartbeat/heartbeat.go:48]: will this result in a log line with repeated text? `failed to send heartbeat(s) due to api error: failed to send heartbeat(s) due to api error: ...`

**Review by alanhamlett — APPROVED** [review]

**Pull request merged**